src/Symfony/Component/HttpKernel/Event/TerminateEvent.php

```diff
@@ -19,6 +23,24 @@
  *
  * @author Jordi Boggiano <j.boggiano@seld.be>
  */
-class TerminateEvent extends PostResponseEvent
+class TerminateEvent extends KernelEvent
 {
+    private $response;
+
+    public function __construct(HttpKernelInterface $kernel, Request $request, Response $response)
+    {
+        parent::__construct($kernel, $request, HttpKernelInterface::MASTER_REQUEST);
+
+        $this->response = $response;
+    }
+
+    /**
+     * Returns the response for which this event was thrown.
+     *
+     * @return Response
```

Comment:
let's use a real type instead? could require making either the class or the method final in 4.4